neo/Wallets/Wallet.cs

```diff
@@ -306,41 +305,36 @@ public Transaction MakeTransaction(List<TransactionAttribute> attributes, IEnume
                 {
                     foreach (var output in cOutputs)
                     {
-                        byte[] script;
-                        using (ScriptBuilder sb2 = new ScriptBuilder())
+                        var balances = new List<(UInt160 Account, BigInteger Value)>();
+                        foreach (UInt160 account in accounts)
                         {
-                            sb2.EmitPush(0);
-                            foreach (UInt160 account in accounts)
+                            byte[] script;
+                            using (ScriptBuilder sb2 = new ScriptBuilder())
                             {
                                 sb2.EmitAppCall(output.AssetId, "balanceOf", account);
-                                sb2.Emit(OpCode.ADD);
+                                script = sb2.ToArray();
                             }
-                            script = sb2.ToArray();
+                            ApplicationEngine engine = ApplicationEngine.Run(script);
+                            if (engine.State.HasFlag(VMState.FAULT)) return null;
+                            balances.Add((account, engine.ResultStack.Pop().GetBigInteger()));
```

Comment:
Launching several scripts instead of a single big one, with additions and queries. Now addition is performed directly on C# balances.